src/blame.rs

```diff
@@ -0,0 +1,539 @@
+  /// class Blame {
+  ///   getHunkByLine(line: number): BlameHunk;
+  /// }
+  /// ```
+  ///
+  /// @param {number} line - Line number to get blame information for (1-based)
+  /// @returns Blame information for the specified line
+  /// @throws If no hunk is found for the line
+  pub fn get_hunk_by_line(&self, line: u32) -> crate::Result<BlameHunk> {
+    let hunk = self
+      .inner
+      .get_line(line as usize)
+      .ok_or_else(|| Error::new(Status::InvalidArg, format!("No blame hunk found for line {}", line)))?;
+
+    Ok(BlameHunk::from(&hunk))
+  }
+
+  #[napi(iterator)]
+  /// Gets all blame hunks as an iterator
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   iter(): Generator<BlameHunk>;
+  /// }
+  /// ```
+  ///
+  /// @returns Iterator of all blame hunks
+  /// @example
+  /// ```ts
+  /// // Using for...of loop
+  /// for (const hunk of blame.iter()) {
+  ///   console.log(hunk.finalCommitId);
+  /// }
+  ///
+  /// // Using spread operator to collect all hunks
+  /// const hunks = [...blame.iter()];
+  /// ```
+  pub fn iter(&self, env: Env) -> crate::Result<BlameHunks> {
+    let inner = Blame {
+      inner: self.inner.clone_with_env(env)?,
+    };
+
+    Ok(BlameHunks {
+      inner,
+      current_index: 0,
+      total_count: self.get_hunk_count(),
+    })
+  }
+
+  #[napi(iterator)]
+  /// Collects blame hunks by scanning file lines as an iterator
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   iterByLine(): Generator<BlameHunk>;
+  /// }
+  /// ```
+  ///
+  /// @returns Iterator of blame hunks collected by line scanning
+  /// @example
+  /// ```ts
+  /// // Using for...of loop
+  /// for (const hunk of blame.iterByLine()) {
+  ///   console.log(hunk.finalCommitId);
+  /// }
+  ///
+  /// // Using spread operator to collect all hunks
+  /// const hunks = [...blame.iterByLine()];
+  /// ```
+  pub fn iter_by_line(&self, env: Env) -> crate::Result<BlameHunksByLine> {
+    let inner = Blame {
+      inner: self.inner.clone_with_env(env)?,
+    };
+
+    Ok(BlameHunksByLine {
+      inner,
+      current_line: 1,
+      processed_hunks: HashSet::new(),
+      total_hunk_count: self.get_hunk_count() as usize,
+    })
+  }
+
+  #[napi]
+  /// Iterates through each hunk and calls the callback function
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   forEachHunk(callback: (hunk: BlameHunk, index: number) => boolean): void;
+  /// }
+  /// ```
+  ///
+  /// @example
+  /// ```ts
+  /// blame.forEachHunk((hunk, index) => {
+  ///   console.log(`Hunk ${index}: ${hunk.finalCommitId}`);
+  ///   return true; // Continue iteration
+  /// });
+  /// ```
+  ///
+  /// @param {Function} callback - Function called for each hunk
+  ///   Return true to continue iteration, false to stop
+  pub fn for_each_hunk(&self, callback: Function<(BlameHunk, u32), bool>) -> crate::Result<()> {
```

Comment:
Now that we've created an Iterator, I think this method could be replaced with an Iterator. What do you think about removing it?